codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyTLV.java

```diff
@@ -0,0 +1,120 @@
+ * under the License.
+ */
+
+package io.netty.handler.codec.haproxy;
+
+import static io.netty.util.internal.ObjectUtil.*;
+
+/**
+ * A Type-Length Value (TLV vector) that can be added to the PROXY protocol
+ * to include additional information like SSL information.
+ *
+ * @see HAProxySSLTLV
+ */
+public class HAProxyTLV {
+
+    private final Type type;
+    private final byte typeByteValue;
+    private final byte[] content;
+
+    /**
+     * The registered types a TLV can have regarding the PROXY protocol 1.5 spec
+     */
+    public enum Type {
+        PP2_TYPE_ALPN,
+        PP2_TYPE_AUTHORITY,
+        PP2_TYPE_SSL,
+        PP2_TYPE_SSL_VERSION,
+        PP2_TYPE_SSL_CN,
+        PP2_TYPE_NETNS,
+        /**
+         * A TLV type that is not officially defined in the spec. May be used for nonstandard TLVs
+         */
+        OTHER;
+
+        /**
+         * Returns the the {@link Type} for a specific byte value as defined in the PROXY protocol 1.5 spec
+         * <p>
+         * If the byte value is not an official one, it will return {@link Type#OTHER}.
+         *
+         * @param byteValue the byte for a type
+         * @return the {@link Type} of a TLV
+         */
+        public static Type typeForByteValue(final byte byteValue) {
+            switch (byteValue) {
+            case 0x01:
+                return PP2_TYPE_ALPN;
+            case 0x02:
+                return PP2_TYPE_AUTHORITY;
+            case 0x20:
+                return PP2_TYPE_SSL;
+            case 0x21:
+                return PP2_TYPE_SSL_VERSION;
+            case 0x22:
+                return PP2_TYPE_SSL_CN;
+            case 0x30:
+                return PP2_TYPE_NETNS;
+            default:
+                return OTHER;
+            }
+        }
+    }
+
+    /**
+     * Creates a new HAProxyTLV
+     *
+     * @param type the {@link Type} of the TLV
+     * @param typeByteValue the byteValue of the TLV. This is especially important if non-standard TLVs are used
+     * @param content the raw content of the TLV as byte array
+     */
+    HAProxyTLV(final Type type, final byte typeByteValue, final byte[] content) {
+        checkNotNull(type, "type");
+        checkNotNull(content, "content");
+
+        this.type = type;
+        this.typeByteValue = typeByteValue;
+        this.content = content;
+    }
+
+    /**
+     * Returns the {@link Type} of this TLV
+     */
+    public Type type() {
+        return type;
+    }
+
+    /**
+     * Returns the raw content of the TLV as byte array
+     */
+    public byte[] content() {
+        return content.clone();
+    }
+
+    /**
+     * Returns the type of the TLV as byte
+     */
+    public byte typeByteValue() {
+        return typeByteValue;
+    }
+
+    /**
+     * Returns <code>true</code> if this TLV encapsulates another TLV (this is typically only the case
+     * for {@link HAProxySSLTLV}s
+     */
+    public boolean encapsulatesOtherTLVs() {
```

Comment:
It doesn't look like this is used ... what is the intended use case for this? in `HAProxyMessage.java` an `instanceof HAProxySSLTLV` check is done to access the `encapsulatedTLVs()` method which presumably is intended to be related to this method? It seems like to get access to what this field determines ... you will have to cast to some other type anyways so I don't understand the value of this method.